pkg/plugin/processor/builtin/webhook/http.go

```diff
@@ -0,0 +1,306 @@
+	// See: sdk.NewReferenceResolver
+	RequestBodyRef string `json:"request.body" default:"."`
+	// ResponseBodyRef specifies to which field should the
+	// response body be saved to.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	ResponseBodyRef string `json:"response.body" default:".Payload.After"`
+	// ResponseStatusRef specifies to which field should the
+	// response status be saved to.
+	// The value of this parameter should be a valid record field reference.
+	// If no value is set, then the response status will NOT be saved.
+	// See: sdk.NewReferenceResolver
+	ResponseStatusRef string `json:"response.status"`
+}
+
+type httpProcessor struct {
+	sdk.UnimplementedProcessor
+
+	logger log.CtxLogger
+	config httpConfig
+
+	requestBodyRef    *sdk.ReferenceResolver
+	responseBodyRef   *sdk.ReferenceResolver
+	responseStatusRef *sdk.ReferenceResolver
+}
+
+func NewWebhookHTTP(l log.CtxLogger) sdk.Processor {
+	return &httpProcessor{logger: l.WithComponent("webhook.httpProcessor")}
+}
+
+func (p *httpProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "webhook.http",
+		Summary: "HTTP webhook processor",
+		Description: `
+A processor that sends an HTTP request to the specified URL, retries on error and 
+saves the response body and, optionally, the response status. 
+`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: httpConfig{}.Parameters(),
+	}, nil
+}
+
+func (p *httpProcessor) Configure(_ context.Context, m map[string]string) error {
+	cfg := httpConfig{}
+	inputCfg := config.Config(m).
+		Sanitize().
+		ApplyDefaults(cfg.Parameters())
+
+	err := inputCfg.Validate(cfg.Parameters())
+	if err != nil {
+		return cerrors.Errorf("invalid configuration: %w", err)
+	}
+
+	if inputCfg["response.body"] == inputCfg["response.status"] {
+		return cerrors.New("invalid configuration: response.body and response.status set to same field")
+	}
+
+	err = inputCfg.DecodeInto(&cfg)
+	if err != nil {
+		return cerrors.Errorf("failed decoding configuration: %w", err)
+	}
+
+	requestBodyRef, err := sdk.NewReferenceResolver(cfg.RequestBodyRef)
+	if err != nil {
+		return cerrors.Errorf("failed parsing request.body %v: %w", p.config.RequestBodyRef, err)
+	}
+	p.requestBodyRef = &requestBodyRef
+
+	responseBodyRef, err := sdk.NewReferenceResolver(cfg.ResponseBodyRef)
+	if err != nil {
+		return cerrors.Errorf("failed parsing response.body %v: %w", p.config.ResponseBodyRef, err)
+	}
+	p.responseBodyRef = &responseBodyRef
+
+	// This field is optional and, if not set, response status won't be saved.
+	if cfg.ResponseStatusRef != "" {
+		responseStatusRef, err := sdk.NewReferenceResolver(cfg.ResponseStatusRef)
+		if err != nil {
+			return cerrors.Errorf("failed parsing response.status %v: %w", p.config.ResponseStatusRef, err)
+		}
+		p.responseStatusRef = &responseStatusRef
+	}
```

Comment:
This is a bit verbose. Unfortunately, adding a decoding hook for the reference resolver didn't help. The reason is that sdk.ReferenceResolver is not acceptable as a field type by `paramgen`.